### ISSUE TITLE
Generated DoneJS app & plugin file structure

### ISSUE THREAD

**Issue by chasenlehara** [donejs/donejs#843]
The DoneJS app and plugin generators include components in `src/` but models go into their own `src/models/` folder. I think it’d be better if the components were put into a `src/components/` folder.

This issue is intended to be a discussion place for any file structure bikeshedding that we should argue over before DoneJS 1.0 is released.

**Comment by phillipskevin** [comment]
There was also discussion about generating models using a modlet structure: https://github.com/donejs/generator-donejs/issues/114.

**Comment by BigAB** [comment]
Not sure where the best place to comment, but I agree with both these changes.

I see only benefit, with no real detriment, to adding these directories to make the file tree more predictable and readable.

**Comment by phillipskevin** [comment]
We could add support for this (put components in components folder) pretty easily.

We could add an option to the supermodel generator to prompt for a folder and default it to `src/models` and add a prompt to the component generator and default it to `src`.

This way nothing would need to change in the guides and users could do what they want right now. And we can change the default folders in a later release.

**Comment by mjstahl** [comment]
Can we also change the `<component>.html` file name to `demo.html`, to go along with `test.html`

**Comment by Alfredo-Delgado** [comment]
Over 80% of code resides in components. While it makes sense for common components used throughout a project to reside in a `common` or `components` directory, having every component in a `components` directory is redundant.

A project is much easier to join if high level components are in `/src` and its dependencies are hierarchically stored beneath them.

e.g. 
```
- src
  - list-page
    - list-page-list
      - list-page-list-item
```

If at some point a component within the hierarchy is found to be useful somewhere else, then it's not much of a burden to move it to a common directory.

The absolute worst case scenario in a complex project is having a `components` directory that's a flat dogpile of directories with equal weight granted to the most and least significant code.

**Comment by kylegifford** [comment]
On Lowe's, we've been using the `src/components` and `src/models` structure and have found it easy to work with. I like not having any component files in the root (`src`) directory, because we've created several other folders there (`connections`, `helpers`, `pages`, `vendor`, etc). Having components in there would muddle things, IMO.

Long story, +1 to `src/components`.

**Comment by adrifolio** [comment]
I agree with the `src/components` idea. In the projects that I have worked on, the base styles are usually stored under "src" but they are not a component, so separating components to its own directory keeps the structure clean.

I also like the idea of being able to enter where a component should be created as part of generating it, but would default to `src/components`. This gives the option of creating the nesting of components that @Alfredo-Delgado described.

For example:

```
src
├─ components
   └── page
       └── list
           └── item
├─ models
├─ styles

**Comment by Alfredo-Delgado** [comment]
Applying the modlet pattern to models isn't as alarming as the prospect of a dogpile of components. If anything, it only tidies up the status quo.

What's most important when dealing with models is recognizing that they are a horizontal concern spanning the entire project, like design, as opposed to disparate strata in vertical feature implementations.

With this in mind, how models are organized should take into account individual model requirements vs. datasource requirements. e.g.
- Datasource
  - connection
  - behaviors
  - fixtures
- Model
  - itself
  - tests

**Comment by matthewp** [comment]
You can do nested components with generator-donejs already. We do this in the PMO guide: `donejs add component restaurant/list pmo-restaurant-list` created a nested component. The change proposed in this issue is putting them in `components/` so the nested version would be `components/restaurant/list` cc @adrifolio @Alfredo-Delgado

**Comment by ccummings** [comment]
+1 to the `src/components` change.

I also agree with @Alfredo-Delgado that a flat folder structure is not ideal so I've used @matthewp approach to add nesting for sub-components in previous projects and it worked out well.

**Comment by matthewp** [comment]
DoneJS 1.0 has been released so file structure changes can't happen until 2.0.
